resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -469,7 +466,7 @@ private[spark] class KubernetesClusterSchedulerBackend(
       (ENV_DRIVER_URL, driverUrl),
       // Executor backend expects integral value for executor cores, so round it up to an int.
       (ENV_EXECUTOR_CORES, math.ceil(executorCores).toInt.toString),
-      (ENV_EXECUTOR_MEMORY, executorMemoryString),
+      (ENV_EXECUTOR_MEMORY, executorMemoryWithOverhead + "m"),
```

Comment:
I don't think we want to do this. The point of the memory with overhead is to make it higher than what we give the JVM. The JVM should only have the exact amount of heap given by `spark.executor.memory`, but the overhead accounts for the fact that the JVM might also allocate memory off-heap, and in Python's case it also covers for the storage of Python's data.

Hence I think #470 does cover everything we need here.

Reply:
@ash211 @mccheah as i said in #467,Right now, the `ENV_EXECUTOR_MEMORY` is **memory** but `ENV_DRIVER_MEMORY` is **memory+overhead** 

If the `ENV_EXECUTOR_MEMORY` should only be memory, as you said above, the `ENV_DRIVER_MEMORY` should also only be memory rather than memory+overhead,is that right?